### DISCUSSION TITLE
Creating spacy doc objects without supplying a model object?

### DISCUSSION THREAD

**Discussion by SecroLoL** [explosion/spaCy#11093] [Help: Coding & Implementations] [answered]
Hi, I'm currently working with the Stanford University based Stanza NLP models and using spaCy's visualization powers. Upon writing some scripts to visualize different languages, it seems that loading  `model = spacy.load('en_core_web_sm')` works perfectly fine when creating a spaCy `Doc` object and visualizing it with `displacy.render()`. To clarify, this works even when the language being visualized is not English. However, loading the model each time as an argument is expensive, so given that it doesn't seem to matter what language I am visualizing, is there a way to make spaCy Doc objects without giving it a model object in the constructor? 

`document_result = Doc(nlp.vocab, words=words, lemmas=lemmas, heads=heads, deps=deps, pos=tags)` where `nlp = spacy.load("en_core_web_sm")`  would be a sample of code to show what I mean. Even when setting the `words` argument to non-English text, the visualization still renders the text perfectly fine, leading to me suspect that spaCy doesn't actually need the `nlp.vocab` object to visualize my text. Therefore, how might I go about saving the expense of loading the spaCy English model and just visualize custom objects? Perhaps something akin to setting `nlp.vocab` to `None` (although this specific example results in an error).

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @SecroLoL ,

In this case, a [blank pipeline](https://spacy.io/api/top-level#spacy.blank) is what you need.  In the case of English, you can do something like: 

```py
import spacy
nlp = spacy.blank("en")
# texts = ...
docs = nlp.pipe(texts)
```
